cocos/core/root.ts

```diff
@@ -649,11 +650,38 @@ export class Root {
      * @param l @en The light to be destroyed @zh 要销毁的光源
      */
     public destroyLight (l: Light) {
+        if (l.scene) {
+            switch (l.type) {
+            case LightType.DIRECTIONAL:
+                l.scene.removeDirectionalLight(l as DirectionalLight);
+                break;
+            case LightType.SPHERE:
+                l.scene.removeSphereLight(l as SphereLight);
+                break;
+            case LightType.SPOT:
+                l.scene.removeSpotLight(l as SpotLight);
+                break;
+            default:
+                break;
+            }
+        }
+        l.destroy();
+    }
+
+    /**
+     * @en recycle the given light to light object pool
+     * @zh 回收指定的光源到对象池
+     * @param l @en The light to be destroyed @zh 要回收的光源
+     */
+    public recycleLight (l: Light) {
```

Comment:
Please find `destroyLight` usage in Components and replace them with `recycleLight`